ext/s3/sink.go

```diff
@@ -124,30 +135,27 @@ func (s3 *S3Sink) process() error {
 
 		wh, ok := s3.writeHandlers[destinationURI]
 		if !ok {
-			// create new s3 write handler
-			targetDestinationURI, err := url.Parse(destinationURI)
-			if err != nil {
-				s3.Logger().Error(fmt.Sprintf("failed to parse destination URI: %s", destinationURI))
-				return errors.WithStack(err)
-			}
-			if targetDestinationURI.Scheme != "s3" {
-				s3.Logger().Error(fmt.Sprintf("invalid scheme: %s", targetDestinationURI.Scheme))
-				return errors.WithStack(err)
-			}
-			// remove object if overwrite is enabled
-			if s3.enableOverwrite {
-				if err := s3.Retry(func() error {
-					return s3.DryRunable(func() error {
-						s3.Logger().Info(fmt.Sprintf("remove object: %s", destinationURI))
-						return s3.client.DeleteObject(s3.Context(), targetDestinationURI.Host, strings.TrimLeft(targetDestinationURI.Path, "/"))
-					})
-				}); err != nil {
-					s3.Logger().Error(fmt.Sprintf("failed to remove object: %s", destinationURI))
+			var sh io.Writer
```

Comment:
as the feature of compression is exist around file related sink, can we have component that can be shared across the sinks?